File: solutions/elixir/isogram/1/lib/isogram.ex
defmodule Isogram do
  @doc """
  Determines if a word or sentence is an isogram
  """
  @spec isogram?(String.t()) :: boolean
  def isogram?(sentence) do
    sentence
    |> String.replace(~r{[-\s]} , "")
    |> String.downcase()
    |> String.graphemes()
    |> Enum.frequencies()
    |> Enum.all?(fn {_, freq} -> freq == 1 end)
  end
end

File: solutions/javascript/isogram/1/isogram.js
export const isIsogram = (word) => {
  word = word.replace(/[ -]/g,'');
  for (var i = 0; i < word.length; i++) {
    for (var j = i + 1; j < word.length; j++) {
      if (word[i].toLowerCase() === word[j].toLowerCase()) {
        return false;
      }
    }
  }
  return true;
};